### PR TITLE
added config check service

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: app/scripts/controllers/overview.js [hunk 1]
```diff
@@ -64,6 +64,8 @@ angular.module('overview', ['stats'])
       'data': [],
     };
 
+    console.log(JSON.stringify(ClusterState, null, 2));
```
Comment (chaudum) [inline app/scripts/controllers/overview.js]: please remove if not necessary any more :)
we could think about a global `DEBUG` state that enables useful logging

File: app/scripts/services/stats.js [hunk 2]
```diff
@@ -206,8 +211,22 @@ angular.module('stats', ['sql', 'health', 'tableinfo'])
       });
     };
 
+    var refreshConfigCheck = function() {
+      // TODO: implement config Check
```
Comment (chaudum) [inline app/scripts/services/stats.js]: already done, right? :)

File: app/views/overview.html [hunk 3]
```diff
@@ -82,4 +82,25 @@
         </div>
     </div><!-- /.row -->
 
+    <div class="row">
```
Comment (chaudum) [inline app/views/overview.html]: show row only if `num checks > 0`

File: app/views/overview.html [hunk 4]
```diff
@@ -82,4 +82,25 @@
...
+
+      <div class="col-md-6 col-sm-6 col-xs-6">
+        <div class="panel">
+          <h1>Configuration Checks</h1>
+          <ol>
+            <li ng-repeat="check in checks" ng-switch on="check.severity">
+              <p>
+                <strong ng-switch-when="1" class="bg-info">Info</strong>
```
Comment (chaudum) [inline app/views/overview.html]: not sure about the wording: Info/Warning/Critical
we can discuss it tormorrow

File: app/views/overview.html [hunk 5]
```diff
@@ -82,4 +82,25 @@
...
+                <strong ng-switch-when="2" class="bg-warning">Warning</strong>
+                <strong ng-switch-when="3" class="btn-danger">Critical</strong>
+                {{check.description}}
+              </p>
+              <strong ng-class="{ 'passed' : check.passed == false }">not passed</strong>
```
Comment (chaudum) [inline app/views/overview.html]: we do only show NOT passed checks, right?

File: app/styles/theme.less [hunk 6]
```diff
@@ -260,6 +269,25 @@ h2, h3 {
...
+  background-color: #dff0d8;
+}
+.bg-info {
+  background-color: #d9edf7;
+}
+.bg-warning {
+  background-color: #fcf8e3;
+}
+.bg-danger {
+  background-color: #f2dede;
```
Comment (chaudum) [inline app/styles/theme.less]: can you use  pre-defined colors?

File: app/views/overview.html [hunk 7]
```diff
@@ -67,7 +67,7 @@
     </div><!-- /.row -->
 
     <div class="row" ng-show="chart.data">
-        <div class="col-md-12">
+        <div ng-class="checks.length > 0 ? 'col-md-6 col-sm-6 col-xs-6' : 'col-md-12 col-sm-12 col-xs-12'">
```
Comment (chaudum) [inline app/views/overview.html]: do it in Angular way: e.g. `ng-class="{strike: deleted, bold: important, 'has-error': error}"`

File: app/scripts/controllers/common.js [hunk 8]
```diff
@@ -35,6 +35,15 @@ var commons = angular.module('common', ['stats', 'udc'])
       $scope.load15 = data.load[2] == '-.-' ? data.load[2] : data.load[2].toFixed(2);
       $scope.version = data.version;
       $scope.docs_url = getDocsUrl(data.version);
+      $scope.checks = data.checks;
+      $scope.config_label = "success";
+      if (data.checks.length > 0) {
+        data.checks.filter(function(obj) {
+          $scope.config_label = (obj.severity === 1) ? "info" : $scope.config_label;
```
Comment (chaudum) [inline app/scripts/controllers/common.js]: you're misusing the filter function for something that can be done with the reduce function

Comment (chaudum) [inline app/scripts/controllers/common.js]: defined the full label name here, e.g. `label-info` instead of doing string concat in the template

**Comment by chaudum** [comment]
screen shot 2015-08-14 at 17 24 45

File: app/scripts/controllers/common.js [hunk 9]
```diff
@@ -2,10 +2,13 @@
 
 var commons = angular.module('common', ['stats', 'udc'])
   .controller('StatusBarController', function ($scope, $log, $location, $sce, ClusterState) {
-    var colorMap = {"good": 'label-success',
-                    "warning": 'label-warning',
-                    "critical": 'label-danger',
-                    '--': 'label-danger'};
+    var HEALTH = ['good', 'warning', 'critical', '--'];
```
Comment (kovrus) [inline app/scripts/controllers/common.js]: should it be "info" instead of "good"

File: app/scripts/controllers/common.js [hunk 10]
```diff
@@ -2,10 +2,13 @@
...
+    var LABELS = ['label-success', 'label-warning', 'label-danger', 'label-danger'];
```
Comment (kovrus) [inline app/scripts/controllers/common.js]: two "label-danger"

File: app/scripts/services/configcheck.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+angular.module('configcheck', [])
+  .factory('ConfigCheck', ['SQLQuery', 'queryResultToObjects', '$q',
+    function (SQLQuery, queryResultToObjects, $q) {
+      var self = {
+        deferred: $q.defer()
+      };
+
+      // table info statement
+      var configCheckStmt = 'SELECT id, severity, description, passed FROM sys.checks WHERE passed = false ORDER BY severity DESC, id';
```
Comment (kovrus) [inline app/scripts/services/configcheck.js]: configcheckstatement - rename

File: app/scripts/services/configcheck.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+      self.executeConfigCheckStmt = function () {
```
Comment (kovrus) [inline app/scripts/services/configcheck.js]: remove config from the var name

File: app/scripts/services/stats.js [hunk 13]
```diff
@@ -1,14 +1,15 @@
 'use strict';
 
 angular.module('stats', ['sql', 'health', 'tableinfo'])
-  .factory('ClusterState', function ($http, $interval, $timeout, $location, $log, SQLQuery, queryResultToObjects, TableList, TableInfo, Health, ShardInfo, $q) {
-    var healthInterval, statusInterval, reachabilityInterval, shardsInterval;
+  .factory('ClusterState', function ($http, $interval, $timeout, $location, $log, SQLQuery, queryResultToObjects, TableList, TableInfo, Health, ShardInfo, ConfigCheck, $q) {
+    var healthInterval, statusInterval, reachabilityInterval, shardsInterval, checkConfigInterval;
```
Comment (kovrus) [inline app/scripts/services/stats.js]: checkConfigInterval - rename

File: app/scripts/controllers/overview.js [hunk 14]
```diff
@@ -130,6 +132,10 @@ angular.module('overview', ['stats'])
       $scope.chart.data = chart.update(data).data();
     };
 
+    $scope.refreshConfigCheck = function refreshConfigCheck() {
+      ClusterState.refreshConfigCheck();
```
Comment (kovrus) [inline app/scripts/controllers/overview.js]: refreshClusterChecks

**Comment by kovrus** [comment]
ack

**Pull request merged**